pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -207,10 +217,46 @@ func getParseFunc(log []byte) (parseFunc, error) {
 	return nil, fmt.Errorf("unsupported log format: %q", log)
 }
 
+// extractStreamFromLog parses a single log line and returns the stream type.
+func extractStreamFromLog(log []byte) (runtimeapi.LogStreamType, error) {
+	var stream runtimeapi.LogStreamType
+
+	if bytes.HasPrefix(log, []byte{'{'}) {
+		// Docker JSON log format
+		var obj struct {
+			Stream runtimeapi.LogStreamType `json:"stream"`
+		}
+		err := json.Unmarshal(log, &obj)
```

Comment:
The json.unmarshal is potentially expensive as well. Is there a way we can do this without deserializing the entire line? It is also expensive from a GC persepective.

Reply:
🤔 I would also like to avoid this deserializing, but the docker log is formatted in JSON, I don't know if there is another robust way to get the stream type from a single line of log.